packages/forge2d/test/dynamics/joints/gear_joint_test.dart

```diff
@@ -0,0 +1,217 @@
+
+    setUp(() {
+      world = World();
+      revoluteRevoluteGearJointDef = GearJointDef()
+        ..bodyA = Body(BodyDef(), world)
+        ..bodyB = Body(BodyDef(), world)
+        ..joint1 = RevoluteJoint(
+          RevoluteJointDef()
+            ..bodyA = Body(BodyDef(), world)
+            ..bodyB = Body(BodyDef(), world),
+        )
+        ..joint2 = RevoluteJoint(
+          RevoluteJointDef()
+            ..bodyA = Body(BodyDef(), world)
+            ..bodyB = Body(BodyDef(), world),
+        );
+      revolutePrismaticGearJointDef = GearJointDef()
+        ..bodyA = Body(BodyDef(), world)
+        ..bodyB = Body(BodyDef(), world)
+        ..joint1 = RevoluteJoint(
+          RevoluteJointDef()
+            ..bodyA = Body(BodyDef(), world)
+            ..bodyB = Body(BodyDef(), world),
+        )
+        ..joint2 = PrismaticJoint(
+          PrismaticJointDef()
+            ..bodyA = Body(BodyDef(), world)
+            ..bodyB = Body(BodyDef(), world),
+        );
+      prismaticRevoluteGearJointDef = GearJointDef()
+        ..bodyA = Body(BodyDef(), world)
+        ..bodyB = Body(BodyDef(), world)
+        ..joint1 = PrismaticJoint(
+          PrismaticJointDef()
+            ..bodyA = Body(BodyDef(), world)
+            ..bodyB = Body(BodyDef(), world),
+        )
+        ..joint2 = RevoluteJoint(
+          RevoluteJointDef()
+            ..bodyA = Body(BodyDef(), world)
+            ..bodyB = Body(BodyDef(), world),
+        );
+      prismaticPrismaticGearJointDef = GearJointDef()
+        ..bodyA = Body(BodyDef(), world)
+        ..bodyB = Body(BodyDef(), world)
+        ..joint1 = PrismaticJoint(
+          PrismaticJointDef()
+            ..bodyA = Body(BodyDef(), world)
+            ..bodyB = Body(BodyDef(), world),
+        )
+        ..joint2 = PrismaticJoint(
+          PrismaticJointDef()
+            ..bodyA = Body(BodyDef(), world)
+            ..bodyB = Body(BodyDef(), world),
+        );
+    });
+
+    group('constructor', () {
+      group(
+        'builds',
+        () {
+          test('when joint1 and joint2 are both RevoluteJoints', () {
+            expect(GearJoint(revoluteRevoluteGearJointDef), isA<GearJoint>());
+          });
+
+          test('when joint1 is RevoluteJoint and joint2 is PrismaticJoint', () {
+            expect(GearJoint(revolutePrismaticGearJointDef), isA<GearJoint>());
+          });
+
+          test('when joint1 is PrismaticJoint and joint2 is RevoluteJoint', () {
+            expect(GearJoint(prismaticRevoluteGearJointDef), isA<GearJoint>());
+          });
+
+          test('when joint1 and joint2 are both PrismaticJoints', () {
+            expect(GearJoint(revoluteRevoluteGearJointDef), isA<GearJoint>());
+          });
+        },
+      );
+
+      test(
+        'throws AssertionError '
+        'when joints are not RevoluteJoint and/or PrismaticJoint',
+        () {
+          // TODO(alestiago): Consider removing UnknownJoint and test for all
```

Comment:
@spydon I've been talking with @felixangelov about this type of matrix tests. Apparently, there's one of these sitting along in the Flutter or Dart codebase. 

Once I have a clear, concise and maintainable solution I'll raise another PR to include tests for the permutation.